main.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) != true
+    end
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) == true
+    end
+    puts output
```

Comment:
:bulb: Let's remove the `puts`. This will makes the method to return the wrong value. See above comment (_2. When a block is given (This have to do with the puts used)_). :point_up_2: